### PR TITLE
Add compressor support

### PR DESCRIPTION
**Pull request by gabriel-samfira**
This adds the option to use an external, multi-threaded service for compression.

### CODE REVIEW

File: coriolis/data_transfer.py [hunk 1]
```diff
@@ -16,20 +104,13 @@ def encode_data(msg_id, path, offset, content, compress=True):
 
     data_len_inflated = len(inflated_content)
 
+    compressed = False
     if compress:
-        data_content = zlib.compress(inflated_content)
+        data_content, compressed = compression_proxy(
+            inflated_content, constants.COMPRESSION_FORMAT_ZLIB)
```
Comment (aznashwan) [inline coriolis/data_transfer.py]: NIT: I suggest we make the compression type configurable just in case

Comment (gabriel-samfira) [reply coriolis/data_transfer.py]: the ```write_data``` binary expects ```zlib``` compression format. So not sure if it's worth making ```encode_data()``` configurable.

Comment (gabriel-samfira) [reply coriolis/data_transfer.py]: ```compression_proxy()``` does accept a compression format type flag.

**Comment by aznashwan** [comment]
LGTM. Merging. Thanks!

**Pull request merged**